### PR TITLE
nerdctl compose: fix `--env-file` (broken since v1.6.0)

### LINKED ISSUE #2566: nerdctl compose --env-file no longer working after version 1.5.0
### Description

In nerdctl 1.5.0, the `--env-file` flag used together with compose was behaving as expected.

In versions 1.6.0, and now 1.6.1 of nerdctl, the `--env-file` flag no longer processes environment variables in the specified file.

I noticed that the new versions stopped complaining about the --env-file being reference to using a full path as opposed to relative path.

### Steps to reproduce the issue

For a compose file (compose-hello.yml) like:

```
version: "3"

services:

  hello:
    build:
      dockerfile: hello.containerfile
      args:
        configfile: "${config}"
        javaopts: "{java_opts_egd} {java_opts_mem_tiny}"
```

and the following hello.env file:

```
configfile="cfg-230917-345678"
```
Building an image from the hello.containerfile:

```
ARG configfile
FROM scratch
COPY ${configfile}.xml /config.xml
```

The following commands:

```
nerdctl compose --env-file hello.env  -f compose-hello.yml -p hello pull
nerdctl compose --env-file hello.env  -f compose-hello.yml -p hello create --build
```

Fails with:

```
WARN[0000] The "config" variable is not set. Defaulting to a blank string. 
```

### Describe the results you received and expected

The expected outcome is for nerdctl compose to create the named project and build the image as instructed by the compose file.

### What version of nerdctl are you using?

1.6.1 (this also fails in 1.6.0, but worked in 1.5.0)

### Are you using a variant of nerdctl? (e.g., Rancher Desktop)

None

### Host information

$ nerdctl info
Client:
 Namespace:	default
 Debug Mode:	false

Server:
 Server Version: v1.7.7
 Storage Driver: overlayfs
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Log: fluentd journald json-file syslog
  Storage: native overlayfs
 Security Options:
  seccomp
   Profile: builtin
  cgroupns
  rootless
 Kernel Version: 5.14.0-284.25.1.el9_2.x86_64
 Operating System: AlmaLinux 9.2 (Turquoise Kodkod)
 OSType: linux
 Architecture: x86_64
 CPUs: 1
 Total Memory: 7.508GiB
 Name: localhost.localdomain
 ID: 43537b23-e94c-41ea-b77b-3f2c1aae1761

WARNING: No cpu cfs period support
WARNING: No cpu cfs quota support
WARNING: No cpu shares support
WARNING: No cpuset support
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled

### PR DESCRIPTION
**Pull request by AkihiroSuda**
Fix #2566

This was a regression in PR #2472 (v1.6.0)

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
@containerd/nerdctl-maintainers 

I'll release v1.6.2 right after merging this

**Review by ktock — APPROVED** [review]

**Pull request merged**
